Source/DataSources/KmlDataSource.js

```diff
@@ -701,7 +701,12 @@ define([
         var color = queryColorValue(node, 'color', namespaces.kml);
 
         var iconNode = queryFirstNode(node, 'Icon', namespaces.kml);
-        var icon = getIconHref(iconNode, dataSource, sourceUri, uriResolver, false, query);
+        var icon;
+        if (!defined(iconNode)) {
```

Comment:
What situation does this address that isn't addressed below?

Reply:
The situation where <IconStyle> tags are empty.  For an example, see simpleEmptyIconStyle.kml.  This differs from the situation where <IconStyle> tag includes a blank <Icon> inside, as in simpleNoIcon.kml.

Google Earth shows yellow placemark in the first situation and no placemark in the second.  This code here is necessary to have yellow placemark show in the first situation.